cogs/utils/settings.py

```diff
@@ -11,27 +14,96 @@ def __init__(self, path=default_path):
             current = dataIO.load_json(self.path)
             if current.keys() != self.default_settings.keys():
                 for key in self.default_settings.keys():
                     if key not in current.keys():
+                        if key == "TOKEN":
+                            old_format = True
                         current[key] = self.default_settings[key]
                         print("Adding " + str(key) +
                               " field to red settings.json")
                 dataIO.save_json(self.path, current)
             self.bot_settings = dataIO.load_json(self.path)
+
         if "default" not in self.bot_settings:
-            self.update_old_settings()
+            self.update_old_settings_v1()
+
+        if old_format:
+            self.update_old_settings_v2()
+
+        self.parse_cmd_arguments()
+
+    def parse_cmd_arguments(self):
+        parser = argparse.ArgumentParser(description="Red - Discord Bot")
+        parser.add_argument("--email")
+        parser.add_argument("--password")
+        parser.add_argument("--token")
+        parser.add_argument("--owner", help="ID of the owner")
+        parser.add_argument("--prefix", "-p", action="append",
+                            help="Global prefix. Can be multiple.")
+        parser.add_argument("--admin-role")
+        parser.add_argument("--mod-role")
+        parser.add_argument("--no-prompt",
+                            action="store_true",
+                            help="Disables console inputs. Features requiring "
+                                 "console interaction could be disabled as a "
+                                 "result")
+        parser.add_argument("--self-bot",
+                            action='store_true',
+                            help="Specifies if Red should log in as selfbot")
+        parser.add_argument("--memory-only",
+                            action="store_true",
+                            help="Arguments passed and future edits to the "
+                                 "settings will not be saved to disk")
+        parser.add_argument("--debug",
+                            action="store_true",
+                            help="Enables debug mode")
+
+        args = parser.parse_args()
+        args_items = vars(args)
+
+        for arg in args_items:
+            if args_items[arg] is None:
+                continue
+            if arg == "token":
+                self.token = args.token
+                self.login_type = "token"
+            elif arg == "email":
+                self.email = args.email
+                self.password = args.password
+                self.login_type = "email"
+            elif arg == "owner":
+                self.owner = args.owner
+            elif arg == "prefix":
+                self.prefixes = sorted(args.prefix, reverse=True)
+            elif arg == "admin_role":
+                self.default_admin = args.admin_role
+            elif arg == "mod_role":
```

Comment:
I swear it made sense when I was looking at how argparse worked. Refactoring :')